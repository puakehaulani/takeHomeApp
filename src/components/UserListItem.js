import { useNavigation } from '@react-navigation/native';
import React, { useContext } from 'react';
import { View, Text, Pressable, Image, StyleSheet } from 'react-native';
import { ThemeContext } from '../context/themeContext'

export const UserListItem = props => {
  const { user } = props;
  const navigation = useNavigation();
  const { theme } = useContext(ThemeContext);

  const styles = {
    shadow: {
      shadowColor: theme === 'dark' ? '#FAF9F6' : '#121212',
      shadowOffset: { width: -2, height: 4 },
      shadowOpacity: 0.2,
      shadowRadius: 3,
      elevation: 5,
    },
    titleText: {
      color: theme === 'dark' ? '#FAF9F6' : '#121212',
      fontSize: 20,
      fontFamily: 'ArialRoundedMTBold',
    },
    subtitleText: {
      color: 'gray',
      fontSize: 14,
      fontFamily: 'ArialMT',
    },
    bodyText: {
      color: theme === 'dark' ? '#FAF9F6' : '#121212',
      fontSize: 12,
      fontFamily: 'ArialMT',
    },
    boldText: {
      color: theme === 'dark' ? '#FAF9F6' : '#121212',
      fontSize: 12,
      fontWeight: 'bold',
      fontFamily: 'ArialMT',
    },
  };

  return (
    <Pressable onPress={() => navigation.navigate('Profile', { user })}>
      <View
        style={[
          {
            flex: 1,
            flexDirection: 'row',
            borderWidth: 1,
            borderRadius: 20,
            borderColor: theme === 'dark' ? '#121212' : '#FAF9F6',
            height: 70,
            margin: 10,
            alignItems: 'center',
            backgroundColor: theme === 'dark' ? '#121212' : '#FAF9F6',
          },
          styles.shadow,
        ]}>
        <View
          style={{
            flex: 1,
            alignItems: 'center',
          }}>
          <Image
            source={{ uri: `https://i.pravatar.cc/150?img=${(user.id + 6)}` }}
            style={{
              width: 60,
              height: 60,
              borderWidth: 1,
              borderColor: theme === 'dark' ? '#121212' : '#FAF9F6',
              borderRadius: 30,
            }}
          />
        </View>
        <View
          style={{
            flex: 2,
            flexDirection: 'column',
          }}>
          <Text style={styles.titleText}>{user.name.replace(/\b(?:Mrs\.? *|Mr\. *|, Ms)\b/g, '')}</Text>
          <Text style={styles.subtitleText}>{user.email}</Text>
        </View>
      </View>
    </Pressable>
  );
};

