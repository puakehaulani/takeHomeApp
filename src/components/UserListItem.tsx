import React from 'react';
import {View, Text, Pressable, StyleSheet} from 'react-native';

export const UserListItem = props => {
  const {user} = props;
  console.log('item', user);
  return (
    <Pressable onPress={() => console.log(user.name)}>
      <View
        style={[
          {
            flex: 1,
            flexDirection: 'row',
            borderWidth: 1,
            borderRadius: 20,
            borderColor: 'white',
            height: 70,
            margin: 10,
            alignItems: 'center',
            backgroundColor: 'white',
          },
          styles.shadow,
        ]}>
        <View
          style={{
            flex: 1,
            alignItems: 'center',
            //    backgroundColor: 'yellow'
          }}>
          <View
            style={[
              {
                width: 60,
                height: 60,
                borderWidth: 1,
                borderColor: 'pink',
                borderRadius: 30,
                backgroundColor: 'lavender',
              },
              styles.shadow,
            ]}
          />
        </View>
        <View
          style={{
            flex: 2,
            flexDirection: 'column',
            //   backgroundColor: 'pink',
          }}>
          <Text>{user.name}</Text>
          <Text>{user.email}</Text>
        </View>
      </View>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  shadow: {
    shadowColor: '#171717',
    shadowOffset: {width: -2, height: 4},
    shadowOpacity: 0.2,
    shadowRadius: 3,
    elevation: 5,
  },
});
