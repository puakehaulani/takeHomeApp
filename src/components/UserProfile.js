import React, { useContext } from 'react';
import {
  View,
  Text,
  Pressable,
  Image,
  StyleSheet,
  Dimensions,
} from 'react-native';
import { ThemeContext } from '../context/themeContext'

export const UserProfile = props => {
  const user = props?.props;
  const width = Dimensions.get('window').width;
  const height = Dimensions.get('window').height;
  const { theme } = useContext(ThemeContext);

  const styles = {
    shadow: {
      shadowColor: theme === 'dark' ? '#FAF9F6' : '#121212',
      shadowOffset: { width: 1, height: 3 },
      shadowOpacity: 0.2,
      shadowRadius: 5,
      elevation: 5,
    },
    titleText: {
      color: theme === 'dark' ? '#FAF9F6' : '#121212',
      fontSize: 40,
      fontFamily: 'ArialRoundedMTBold',
    },
    subtitleText: {
      color: 'gray',
      fontSize: 20,
      fontFamily: 'ArialMT',
      marginBottom: 10,
    },
    bodyText: {
      color: theme === 'dark' ? '#FAF9F6' : '#121212',
      fontSize: 18,
      fontFamily: 'ArialMT',
      marginBottom: 10,
    },
    boldText: {
      color: theme === 'dark' ? '#FAF9F6' : '#121212',
      fontSize: 16,
      fontWeight: 'bold',
      fontFamily: 'ArialMT',
    },
    smallTitleText: {
      color: 'gray',
      fontSize: 15,
      fontFamily: 'ArialMT',
      marginBottom: 5,
    },
  }

  return (
    <View style={{ height: '100%' }}>
      <View
        style={{
          justifyContent: 'center',
          alignItems: 'center',
          marginTop: 10,
          marginBottom: 200,
        }}>
        <Image
          source={{ uri: `https://i.pravatar.cc/150?img=${(user.id + 6)}` }}
          style={{
            width: 160,
            height: 160,
            borderWidth: 1,
            borderColor: 'transparent',
            borderRadius: 80,
            zIndex: 1,
          }}
        />
        <View
          style={[
            {
              borderWidth: 1,
              borderRadius: 20,
              borderColor: theme === 'dark' ? "#121212" : 'white',
              width: 0.9 * width,
              alignItems: 'center',
              backgroundColor: theme === 'dark' ? "#121212" : 'white',
              zIndex: 0.1,
              paddingTop: 120,
              paddingBottom: 10,
              position: 'absolute',
              left: Dimensions.get('window').width * 0.05,
              top: Dimensions.get('window').height * 0.06,
            },
            styles.shadow,
          ]}>
          <View style={{ flexDirection: 'column' }}>
            <View style={{ alignItems: 'center' }}>
              <Text style={[styles.titleText, { textAlign: 'center' }]}>{user.name.replace(/\b(?:Mrs\.? *|Mr\. *|, Ms)\b/g, '')}</Text>
              <Text style={styles.subtitleText}>@{user.username}</Text>
            </View>
          </View>
          <View style={{ flexDirection: 'row' }}>
            <View style={{ alignItems: 'center' }}>
              <Text style={styles.smallTitleText}>company</Text>
              <Text style={styles.boldText}>{user.company.name}</Text>
            </View>
            <Text
              style={{
                marginHorizontal: 10,
                fontSize: 40,
                color: 'gray',
                fontFamily: 'AvenirNext-UltraLight',
              }}>
              |
            </Text>
            <View style={{ alignItems: 'center' }}>
              <Text style={styles.smallTitleText}>website</Text>
              <Text style={styles.boldText}>{user.website}</Text>
            </View>
          </View>
        </View>
      </View>
      <View style={{ marginHorizontal: 25, marginTop: 10 }}>
        <Text style={styles.smallTitleText}>email</Text>
        <Text style={styles.bodyText}>{user.email}</Text>
        <Text style={styles.smallTitleText}>phone</Text>
        <Text style={styles.bodyText}>{user.phone.replace(/\./g, '-').replace(/\(/g, '').replace(/\)/g, '-')}</Text>
        <Text style={styles.smallTitleText}>address</Text>
        <Text style={styles.bodyText}>
          {user.address.suite} {user.address.street}
        </Text>
        <Text style={styles.bodyText}>
          {user.address.city} {user.address.zipcode.substr(0, 5)}
        </Text>
      </View>
    </View>
  );
};
