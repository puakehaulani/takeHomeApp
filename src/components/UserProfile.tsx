import React from 'react';
import {
  View,
  Text,
  Pressable,
  Image,
  StyleSheet,
  Dimensions,
} from 'react-native';

export const UserProfile = () => {
  const width = Dimensions.get('window').width;
  const height = Dimensions.get('window').height;
  return (
    <>
      <View
        style={{
          justifyContent: 'center',
          alignItems: 'center',
        }}>
        <Image
          source={{uri: `https://i.pravatar.cc/150?img=4`}}
          style={{
            width: 160,
            height: 160,
            borderWidth: 1,
            borderColor: 'transparent',
            borderRadius: 80,
            zIndex: 1,
          }}
        />
        <View
          style={[
            {
              borderWidth: 1,
              borderRadius: 20,
              borderColor: 'white',
              width: 0.9 * width,
              alignItems: 'center',
              backgroundColor: 'white',
              zIndex: 0.1,
              paddingTop: 120,
              position: 'absolute',
              left: Dimensions.get('window').width * 0.05,
              top: Dimensions.get('window').height * 0.06,
            },
            styles.shadow,
          ]}>
          <View style={{flexDirection: 'column'}}>
            <View style={{alignItems: 'center'}}>
              <Text style={styles.titleText}>frank dog</Text>
              <Text style={styles.subtitleText}>@frankDOG</Text>
            </View>
          </View>
          <View style={{flexDirection: 'row'}}>
            <View style={{alignItems: 'center'}}>
              <Text style={styles.smallTitleText}>company</Text>
              <Text style={styles.boldText}>frankDog & co.</Text>
            </View>
            <Text
              style={{
                marginHorizontal: 10,
                fontSize: 40,
                color: 'gray',
                fontFamily: 'AvenirNext-UltraLight',
              }}>
              |
            </Text>
            <View style={{alignItems: 'center'}}>
              <Text style={styles.smallTitleText}>website</Text>
              <Text style={styles.boldText}>frankiedog.com</Text>
            </View>
          </View>
        </View>
      </View>
      <View style={{marginHorizontal: 10, marginTop: 160}}>
        <Text style={styles.smallTitleText}>email</Text>
        <Text style={styles.bodyText}>bork@email.com</Text>
        <Text style={styles.smallTitleText}>phone</Text>
        <Text style={styles.bodyText}>808-344-7521</Text>
        <Text style={styles.smallTitleText}>address</Text>
        <Text style={styles.bodyText}>1234 Dogfarm Ln Seattle WA 98114</Text>
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  shadow: {
    shadowColor: '#171717',
    shadowOffset: {width: 1, height: 3},
    shadowOpacity: 0.2,
    shadowRadius: 5,
    elevation: 5,
  },
  titleText: {
    color: 'black',
    fontSize: 40,
    fontFamily: 'ArialRoundedMTBold',
  },
  subtitleText: {
    color: 'gray',
    fontSize: 20,
    fontFamily: 'ArialMT',
    marginBottom: 10,
  },
  bodyText: {
    color: 'black',
    fontSize: 18,
    fontFamily: 'ArialMT',
    marginBottom: 10,
  },
  boldText: {
    color: 'black',
    fontSize: 16,
    fontWeight: 'bold',
    fontFamily: 'ArialMT',
  },
  smallTitleText: {
    color: 'gray',
    fontSize: 15,
    fontFamily: 'ArialMT',
    marginBottom: 5,
  },
});
