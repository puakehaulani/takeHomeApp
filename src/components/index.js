export * from './UserList'
export * from './UserListItem'
export * from './UserProfile'