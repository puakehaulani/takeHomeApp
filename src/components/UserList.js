import React from 'react';
import { UserListItem } from './UserListItem';

export const UserList = props => {
  const userData = props.props;
  const renderList = userData?.map(user => (
    <UserListItem user={user} key={user.id} />
  ));
  return renderList;
};
