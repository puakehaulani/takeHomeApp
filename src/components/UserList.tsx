import React from 'react';
import {Text} from 'react-native';
import {UserListItem} from './UserListItem';

export const UserList = props => {
  const userData = props.props;
  const renderList = userData?.map(user => (
    <UserListItem user={user} key={user.id} />
  ));
  return renderList;
};
