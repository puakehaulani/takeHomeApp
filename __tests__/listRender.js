import React from 'react';
import renderer from 'react-test-renderer';
import { UserList } from '../src/components';

test('userList renders correctly', () => {
    const tree = renderer.create(<UserList />).toJSON();
    expect(tree).toMatchSnapshot();
});
