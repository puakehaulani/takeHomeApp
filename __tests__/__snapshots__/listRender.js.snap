// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list renders correctly 1`] = `null`;

exports[`userList renders correctly 1`] = `null`;
